spec/requests/courses_spec.rb

```diff
@@ -22,95 +22,154 @@
   let(:subject) { FactoryBot.create(:subject) }
   let(:valid_attributes) { { name: "Biology 101" } }
   let(:invalid_attributes) { { name: nil } }
-
-  before { login_as(user, scope: :user) }
+  let(:login) { login_as(user, scope: :user) }
 
   describe "GET /show" do
-    it "renders a successful response" do
-      course = FactoryBot.create(:course)
-      get subject_course_url(subject, course)
-      expect(response).to be_successful
+    context "without logging in" do
+      it "redirects to users/sign_in page" do
+        course = FactoryBot.create(:course)
+        get subject_course_url(subject, course)
+        expect(response).to redirect_to("/users/sign_in")
+      end
+    end
+
+    context "with logging in" do
+      it "renders a successful response" do
+        login
```

Comment:
issue: It would be better to include the devise test helpers so you can use the `sign_in` method. 
https://github.com/pulibrary/abid/blob/main/spec/support/devise.rb 
https://github.com/pulibrary/abid/blob/main/spec/requests/welcome_spec.rb#L17